### DISCUSSION TITLE
issue: can not get value after setValue

### DISCUSSION THREAD

**Discussion by hungdev** [react-hook-form/react-hook-form#8483] [Q&A] [answered]
### Version Number

7.31.3

### Steps to reproduce

I use setValue in useEffect, but i can get it from value from field in render to fill it to input tag.
Please look my example: https://codesandbox.io/s/fast-monad-u8dvx2?file=/src/Input.js

### Expected behaviour

the value that used setvalue will be dumped into the input tag

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sdcorp** [comment, marked as answer]
@hungdev I think you overcomplicate things. You may fill default values to your form by passing them to the hook as option or individually as a prop to the form element

```useForm({ defaultValues: { name: 'value'} })```

**Reply by bluebill1049** [reply]
pretty much the answer above, or use reset API to reset the entire form. (thank you @sdcorp )

**Reply by hungdev** [reply]
Thank for your quick reply. @sdcorp and @bluebill1049 .
No, i dont want to use defaultValues, because in some cases i have to calculate or wait data before setting default values ( for example is getting data from api,....), so i dont want to use defaultValues. But if i do that, why doesn't it return value. 
I tried with reset api, but if i have 2 fields, reset only works for one, it doesnt work for remaining field. Please take a look my example and try it.

**Reply by hungdev** [reply]
@bluebill1049 is there anyway to do it?

**Reply by bluebill1049** [reply]
when you are using the `reset`, you are supposed to provide the entire form values instead of just a single field, you can probably combine with `getValues()`

```tsx
reset({
  ...getValues(),
  newData: 'test'
})
```

**Reply by hungdev** [reply]
@bluebill1049  when i use 
```
reset({
  ...getValues(),
  newData: 'test'
})
```
it still gets error. Reset will not apply for last field.
you can see example: https://codesandbox.io/s/fast-monad-u8dvx2?file=/src/Input.js
but if i change something ( for example remove space,..) and there are have hot reload, then it will fill data to field. is this bug?
